libp2p/transports/quictransport.nim

```diff
@@ -150,9 +150,28 @@ type QuicTransport* = ref object of Transport
   privateKey: PrivateKey
   connections: seq[P2PConnection]
 
-func new*(_: type QuicTransport, u: Upgrade, privateKey: PrivateKey): QuicTransport =
+proc makeCertificateVerifier(): CertificateVerifier =
+  proc certificateVerifier(certificatesDer: seq[seq[byte]]): bool =
+    if certificatesDer.len != 1:
+      trace "CertificateVerifier: expected one certificate in the chain",
+        cert_count = certificatesDer.len
+      return false
+
+    let cert =
+      try:
+        parse(certificatesDer[0])
+      except CertificateParsingError as e:
+        trace "CertificateVerifier: failed to parse certificate", msg = e.msg
+        return false
+
+    return cert.verify()
```

Comment:
I missed this in the PR that introduces `verify*`, but we should also validate the expiration date of the certificate:
```
The certificate MUST have NotBefore and NotAfter fields set such that the certificate is valid at the time it is received by the peer. When receiving the certificate chain, an endpoint MUST check these conditions and abort the connection attempt if (a) the presented certificate is not yet valid, OR (b) if it is expired. 
```